src/main/java/seedu/address/model/project/UniqueProjectList.java

```diff
@@ -0,0 +1,142 @@
+package seedu.address.model.project;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.util.Iterator;
+import java.util.List;
+
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
+import seedu.address.model.project.exceptions.DuplicateProjectException;
+import seedu.address.model.project.exceptions.ProjectNotFoundException;
+
+/**
+ * A list of projects that enforces uniqueness between its elements and does not allow nulls.
```

Comment:
I love how your comments are so detailed btw! :)

Reply:
Not my comments, copied from person haha